# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookieManager.idl
# Commit: 08acd8256071
# Full Hash: 08acd8256071ad86345dec3fa66767635b07a4b1
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-09 04:16:57
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 3ae9463ee988 -r 08acd8256071 netwerk/cookie/nsICookieManager.idl
--- a/netwerk/cookie/nsICookieManager.idl	Wed Jan 08 19:25:58 2025 +0000
+++ b/netwerk/cookie/nsICookieManager.idl	Wed Jan 08 19:25:59 2025 +0000
@@ -5,6 +5,7 @@
 
 #include "nsISupports.idl"
 #include "nsICookie.idl"
+#include "nsIThirdPartyCookieBlockingExceptionListService.idl"
 
 %{ C++
 namespace mozilla {
@@ -295,4 +296,19 @@
    * Retrieves all the cookies that were created on or after aSinceWhen, order
    * by creation time */
   Array<nsICookie> getCookiesSince(in int64_t aSinceWhen);
+
+
+  /**
+   * Adds a list of exceptions to the third party cookie blocking exception
+   * list.
+   */
+  void addThirdPartyCookieBlockingExceptions(
+    in Array<nsIThirdPartyCookieExceptionEntry> aExcpetions);
+
+  /**
+   * Removes a list of exceptions from the third party cookie blocking
+   * exception list.
+   */
+  void removeThirdPartyCookieBlockingExceptions(
+    in Array<nsIThirdPartyCookieExceptionEntry> aExceptions);
 };